Mobile Release v1.91.0

SiobhyB opened the pull request:
## Description
Release 1.91.0 of the react-native-editor and Gutenberg-Mobile.

For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/5564

SiobhyB committed:
Release script: Update react-native-editor version to 1.91.0

SiobhyB committed:
Release script: Update with changes from 'npm run core preios'

SiobhyB committed:
Update CHANGELOG to reflect changes in release

fluiddot commented on packages/react-native-editor/CHANGELOG.md:
I checked the PRs with the label `Mobile App - i.e. Android or iOS` that got merged after the release `1.90.0` ([reference](https://github.com/WordPress/gutenberg/pulls?q=is%3Apr+is%3Aclosed+label%3A%22Mobile+App+-+i.e.+Android+or+iOS%22)), and noticed that some of them are not included in the changelog:

<img width="1203" alt="Screenshot 2023-03-16 at 15 13 44" src="https://user-images.githubusercontent.com/14905380/225644667-c170a959-235c-4496-823c-e44bc5d2482c.png">

I'm wondering if we could add them, @SiobhyB WDYT?

fluiddot replied:
I'd like to note that https://github.com/WordPress/gutenberg/pull/48776 was superseded by https://github.com/WordPress/gutenberg/pull/48802, so we could add a single entry for both.

SiobhyB replied:
@fluiddot, I'd opted not to include them as they're not user-facing, but could certainly add them if you feel they should be included. :)

fluiddot replied:
Yep, good point. Both PRs although they introduce changes in the user interface shouldn't be noticeable to the user so let's not include them 👍. Hence, the changelog is ok as is, thanks!

fluiddot reviewed (approved):
LGTM 🎊 !

I just added a minor comment about potential new entries in the changelog but shouldn't block the PR.

SiobhyB committed:
Update CHANGELOG to remove reverted change

SiobhyB committed:
[RNMobile] Prevent blocks being un-wantingly replaced with newly added blocks (#49154)

The changes in logic surrounding dictation handling in #49056 led to a regression where currently selected blocks are replaced with newly added blocks. This PR addresses that issue by reverting the changes to the dictation logic.
